rosbag2_tests/test/rosbag2_tests/test_rosbag2_play_end_to_end.cpp

```diff
@@ -95,8 +151,9 @@ TEST_P(PlayEndToEndTestFixture, play_end_to_end_test) {
         Field(
           &test_msgs::msg::Arrays::string_values,
           ElementsAre("Complex Hello1", "Complex Hello2", "Complex Hello3")))));
+  stop_execution(process_id);
+  cleanup_process_handle.cancel();
```

Comment:
Do we need this at all?  Can't we just let `cleanup_process_handle` go out of scope and do this for us?

(same question a few times below as well)

Reply:
The rationale for calling `stop_execution(process_id);` explicitly is because there are inside of it a bunch or gtest checks which might fail and it would be better to have a clean backtrace where from it fails rather then generic failure from `scope_exit`. When there are only one call for start/stop execution it might be not a big deal and would be obviously that error happened when leaving the test case. But in the case when multiple calls for start/stop execution in on test e.g. `play_filters_by_topic` it might be not obvious to understand in what exact scenario it was failing.

For the sake of consistency and chances that someone will add more start/stop execution calls in the test I decided to keep explicit call for stop.

Reply:
> The rationale for calling `stop_execution(process_id);` explicitly is because there are inside of it a bunch or gtest checks which might fail and it would be better to have a clean backtrace where from it fails rather then generic failure from `scope_exit`

All right, that is a good reason to do it explicitly.